# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/pine-uclibc.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- pine4.58/pico/osdep/unix.orig	2005-01-22 22:34:02.000000000 -0800
+++ pine4.58/pico/osdep/unix	2005-01-22 22:34:50.000000000 -0800
@@ -2644,7 +2644,7 @@
 }
 
 
-
+#include <string.h>
 /*
  * errstr - return system error string corresponding to given errno
  *          Note: strerror() is not provided on all systems, so it's 
@@ -2654,7 +2654,7 @@
 errstr(err)
 int err;
 {
-    return((err >= 0 && err < sys_nerr) ? (char *)sys_errlist[err] : NULL);
+    return((err >= 0) ? strerror(err) : NULL);
 }
 
 
--- pine4.58/pine/osdep/err_desc.orig	2005-01-22 22:51:32.000000000 -0800
+++ pine4.58/pine/osdep/err_desc	2005-01-22 22:52:06.000000000 -0800
@@ -1,3 +1,4 @@
+#include <string.h>
 /*----------------------------------------------------------------------
        Return string describing the error
 
@@ -11,8 +12,8 @@
 {
     static char buffer[50+1];
 
-    if(errnumber >= 0 && errnumber < sys_nerr)
-      sprintf(buffer, "%.*s", 50, sys_errlist[errnumber]);
+    if(errnumber >= 0 && strerror(errnumber))
+      sprintf(buffer, "%.*s", 50, strerror(errnumber));
     else
       sprintf(buffer, "Unknown error #%d", errnumber);
 
--- pine4.58/pine/rpload.c.orig	2005-01-22 23:00:36.000000000 -0800
+++ pine4.58/pine/rpload.c	2005-01-22 23:12:55.000000000 -0800
@@ -649,7 +649,7 @@
 {
     static char buffer[50];
 
-    strcpy(buffer, sys_errlist[err]);
+    strcpy(buffer, (strerror(err)?strerror(err):"Unknown error"));
 
     return((char *)buffer);
 }
--- pine4.58/pine/rpdump.c.orig	2005-01-22 23:25:53.000000000 -0800
+++ pine4.58/pine/rpdump.c	2005-01-22 23:26:36.000000000 -0800
@@ -418,7 +418,7 @@
 {
     static char buffer[50];
 
-    strcpy(buffer, sys_errlist[err]);
+    strcpy(buffer, (strerror(err)? strerror(err) : "Unknown error"));
 
     return((char *)buffer);
 }
